### PR TITLE
Transact

### PR DESCRIPTION
**Pull request by ncalexan**
@rnewman this isn't ready to land, but you can at least see the approach coming together.

I'll start knocking up the schema support after lunch.

### CODE REVIEW

File: src/datomish/query.cljc [hunk 1]
```diff
@@ -127,10 +133,10 @@
 (defn apply-elements-to-context [context elements]
   (assoc context :elements elements))
 
-(defn patterns->context
-  "Turn a sequence of patterns into a Context."
-  [patterns]
-  (reduce apply-pattern-to-context (make-context) patterns))
+(defn patterns->into-context
```
Comment (rnewman) [inline src/datomish/query.cljc]: These should be `apply-patterns-to-context`, I think. Certainly not both `->` ("transform") and `into`.

**Comment by rnewman** [comment]
@ncalexan: I cherry-picked a bunch of these commits onto master so that I can safely work on `query.cljc`.

**Comment by rnewman** [comment]
I ported all of this work on top of #16. Tests pass.

**Comment by ncalexan** [comment]
This is happening over in #18.

**Pull request closed without merging**